src/scripts/services/okta-service.ts

```diff
@@ -74,6 +75,55 @@ export const validateCredentials = (
   );
 };
 
+/**
+ * Retrieves logs from the okta server. Unlike other service methods, this method does not filter the Okta specific types
+ * down into a domain specific one because the primary use case of this method is to provide flexible display to the end user
+ * in conjunction with another tool like `jq`.
+ *
+ * @param client the Okta client to use to retrieve the logs.
+ * @param limit the maximum number of logs to retrieve.
+ * @param query the query to filter the logs by (e.g. OIDC).
+ * @param filter the filter to filter the logs by (e.g. eventType eq "user.session.start").
+ * @param since the date from which to retrieve logs.
+ * @param until the date until which to retrieve logs.
+ * @returns the logs retrieved from the Okta server.
+ */
+export const retrieveLogs = (
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  client: okta.Client,
+  limit: number,
+  query?: string,
+  filter?: string,
+  since?: ReadonlyDate,
+  until?: ReadonlyDate
+) => {
+  return pipe(
+    TE.tryCatch(
+      // eslint-disable-next-line functional/no-return-void
+      async () => {
+        // eslint-disable-next-line functional/no-let, functional/prefer-readonly-type
```

Comment:
I think this `functional/no-let` may have been left over from different code?

Reply:
It is. Fixed.